### PR TITLE
Issue #6 chore: T&C - User shouldn't redirect back to T&C page after accepting T&C.

### PR DESCRIPTION
**Pull request by samadhansalunkhe**
- [x] Have you followed the [Contribution Guidenlines](https://github.com/ekstep/Community-Portal-Joomla/wiki/Contribution-guidenlines)?
- [x] Have you done PHPCS?
- [ ] Have you done JSHINT? - NA
- [x] Have you checked for types in language files (if any)?
- [x] Have you checked Acceptance criteria?

Worked in below areas:
1. The user shouldn't redirect back to the T&C page after accepting T&C. - Fixed from Amit U
2. Global T&C - Fixed start date issue.
3. Removed deprecated methods.
4. Removed htmlspecialchar method from T&C view file. - If the editor is **TyniMCE** then it showing HTML tag as it is but its secured from HTML and XSS attack.
5. Added **$db->quote** names in getListQuery method.

### CODE REVIEW

File: com_tc/tc.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <creationDate>2016-10-24</creationDate>
     <copyright>2016 Parth Lawate</copyright>
```
Comment (manojLondhe) [inline com_tc/tc.xml]: change date, copyright, author, version etc

File: com_tc/site/views/content/view.html.php [hunk 2]
```diff
@@ -36,18 +36,33 @@ public function display($tpl = null)
...
+
+				return false;
+			}
+
+			$this->termsandconditions = $model->getItem($this->tc_id);
+			parent::display($tpl);
+		}
+		else
+		{
+			$app->redirect(JRoute::_(JURI::base()));
```
Comment (manojLondhe) [inline com_tc/site/views/content/view.html.php]: JURI => JUri

File: com_tc/site/views/content/tmpl/default.php [hunk 3]
```diff
@@ -36,7 +36,7 @@ function validateform(form)
 			<br>
 			<div class="">
 				<?php
-						echo nl2br(htmlspecialchars($this->termsandconditions->content, ENT_COMPAT, 'UTF-8'));
+						echo nl2br($this->termsandconditions->content);
```
Comment (manojLondhe) [inline com_tc/site/views/content/tmpl/default.php:40]: why this change needed?

File: com_tc/tc.xml (hunk 1 above)
Comment (samadhansalunkhe) [reply com_tc/tc.xml]: Done

File: com_tc/site/views/content/view.html.php (hunk 2 above)
Comment (samadhansalunkhe) [reply com_tc/site/views/content/view.html.php]: Done

File: com_tc/site/views/content/tmpl/default.php (hunk 3 above)
Comment (samadhansalunkhe) [reply com_tc/site/views/content/tmpl/default.php:40]: If the T&C has created using TyniMCE editor then it showing `<p>` tag for text sentences on view page of T&C but its working fine with MD editor if the code contains htmlspecialchar method.

I tested it with removing htmlspecialchar() method for both editors also checked HTML and XSS attacks and its working fine.
